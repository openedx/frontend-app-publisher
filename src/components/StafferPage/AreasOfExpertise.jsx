import React from 'react';
import PropTypes from 'prop-types';
import { Field, FieldArray } from 'redux-form';

import RenderInputTextField from '../RenderInputTextField';
import RemoveButton from '../RemoveButton';
import FieldLabel from '../FieldLabel';

class AreasOfExpertise extends React.Component {
  constructor(props) {
    super(props);

    this.handleRemove = this.handleRemove.bind(this);
  }

  handleRemove(index) {
    this.props.fields.remove(index);
  }

  render() {
    const { fields } = this.props;

    return (
      <div className="areas-of-expertise mb-3">
        <ul className="list-group p-0 m-0 container-fluid">
          {fields.map((expertise, index) => (
            <li className="area-of-expertise list-group-item row d-flex align-items-center px-0 mx-0" key={expertise}>
              <div className="col-11">
                <Field
                  name={`${expertise}.value`}
                  component={RenderInputTextField}
                  type="text"
                  label={<FieldLabel text="Area of expertise" required />}
                  required
                />
              </div>
              <input
                name={`${expertise}.id`}
                type="hidden"
              />
              <RemoveButton
                className="col-1"
                label="Remove area of expertise"
                onRemove={this.handleRemove}
                targetFieldNumber={index}
              />
            </li>
          ))}
        </ul>
        <button
          type="button"
          className="btn btn-outline-primary js-add-button mt-2"
          onClick={() => fields.push({})}
        >
          Add area of expertise
        </button>
      </div>
    );
  }
}

AreasOfExpertise.propTypes = {
  fields: PropTypes.instanceOf(FieldArray).isRequired,
};

export default AreasOfExpertise;
