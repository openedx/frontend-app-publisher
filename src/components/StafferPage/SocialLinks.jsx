import React from 'react';
import PropTypes from 'prop-types';
import { Field, FieldArray } from 'redux-form';

import RenderInputTextField from '../RenderInputTextField';
import RenderSelectField from '../RenderSelectField';
import RemoveButton from '../RemoveButton';
import FieldLabel from '../FieldLabel';

class SocialLinks extends React.Component {
  constructor(props) {
    super(props);

    this.handleRemove = this.handleRemove.bind(this);
  }

  getSocialOptions() {
    return [
      { label: 'Please select a type', value: '' },
      { label: 'Facebook', value: 'facebook' },
      { label: 'Twitter', value: 'twitter' },
      { label: 'Blog', value: 'blog' },
      { label: 'Other', value: 'others' },
    ];
  }

  handleRemove(index) {
    this.props.fields.remove(index);
  }

  render() {
    const { fields } = this.props;

    return (
      <div className="social-links mb-3">
        <ul className="list-group p-0 m-0 container-fluid">
          {fields.map((link, index) => (
            <li className="social-link list-group-item row d-flex align-items-center px-0 mx-0" key={link}>
              <div className="col-4">
                <Field
                  name={`${link}.type`}
                  component={RenderSelectField}
                  options={this.getSocialOptions()}
                  label={<FieldLabel text="Type" required />}
                  required
                />
              </div>
              <div className="col-3">
                <Field
                  name={`${link}.title`}
                  component={RenderInputTextField}
                  type="text"
                  label={<FieldLabel text="Link Text" />}
                />
              </div>
              <div className="col-4">
                <Field
                  name={`${link}.url`}
                  component={RenderInputTextField}
                  type="url"
                  label={<FieldLabel text="URL" required />}
                  required
                />
              </div>
              <input
                name={`${link}.id`}
                type="hidden"
              />
              <RemoveButton
                className="col-1 m-auto"
                label="Remove social link"
                onRemove={this.handleRemove}
                targetFieldNumber={index}
              />
            </li>
          ))}
        </ul>
        <button
          type="button"
          className="btn btn-outline-primary js-add-button mt-2"
          onClick={() => fields.push({})}
        >
          Add social link
        </button>
      </div>
    );
  }
}

SocialLinks.propTypes = {
  fields: PropTypes.instanceOf(FieldArray).isRequired,
};

export default SocialLinks;
