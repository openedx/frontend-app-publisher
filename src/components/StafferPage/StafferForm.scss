.create-staffer-form {
  .staffer-image {
    .uploaded-image {
      max: {
        // These match the 'medium' size in course-discovery's model definition for the image field
        width: 110px;
        height: 110px;
      }
    }
  }

  // react-select specific styles
  .react-select__menu {
    z-index: 5;
  }

  .react-select__control {
    cursor: text;
  }

  .react-select__control.danger {
    border-color: red;
    box-shadow: none;
  }

  .react-select__control--menu-is-open{
    .react-select__single-value{
      color: #D3D3D3
    }
  }
}
