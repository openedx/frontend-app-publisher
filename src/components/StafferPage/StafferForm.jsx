import React from 'react';
import { Field, FieldArray, reduxForm } from 'redux-form';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import ActionButton from '../ActionButton';
import AreasOfExpertise from './AreasOfExpertise';
import SocialLinks from './SocialLinks';
import ImageUpload from '../ImageUpload';
import RenderInputTextField from '../RenderInputTextField';
import ReduxFormCreatableSelect from '../ReduxFormCreatableSelect';
import RichEditor from '../RichEditor';
import FieldLabel from '../FieldLabel';
import ButtonToolbar from '../ButtonToolbar';
import { basicValidate, handleStafferOrCreateFormFail } from '../../utils/validation';

const BaseStafferForm = ({
  handleSubmit,
  pristine,
  isSaving,
  isCreateForm,
  sourceInfo: { referrer },
  cancelStafferInfo,
  organizationName,
  publisherOrganizations,
}) => {
  const formControlDisabled = pristine || isSaving;
  const allOrganizations = organizationName
    ? [...publisherOrganizations.map(org => org.name), organizationName]
    : [...publisherOrganizations.map(org => org.name)];
  const allUniqueOrganizations = [...new Set(allOrganizations)];

  return (
    <div className="create-staffer-form">
      <form onSubmit={handleSubmit}>
        <Field
          name="profile_image.medium.url"
          component={ImageUpload}
          label={(
            <FieldLabel
              id="image.label"
              text="Image"
              helpText={(
                <div>
                  <p>Image Requirements:</p>
                  <ul>
                    <li>The image dimensions must be 110Ã—110.</li>
                    <li>The image size must be less than 256KB.</li>
                  </ul>
                </div>
              )}
              extraText="Image must be 110x110 pixels in size."
            />
          )}
          id="profile_image"
          maxImageSizeKilo={256}
          requiredWidth={110}
          requiredHeight={110}
          className="staffer-image"
          required={isCreateForm}
          validate={basicValidate}
        />
        <Field
          name="given_name"
          component={RenderInputTextField}
          type="text"
          label={<FieldLabel text="First name" />}
          required
        />
        <Field
          name="family_name"
          component={RenderInputTextField}
          type="text"
          label={<FieldLabel text="Last name" />}
          required
        />
        <Field
          name="position.title"
          component={RenderInputTextField}
          type="text"
          label={(
            <FieldLabel
              id="title.label"
              text="Title"
              helpText={(
                <div>
                  <p>Instructor&apos;s title at your organization.</p>
                  <p><b>Examples:</b></p>
                  <ul>
                    <li>Professor</li>
                    <li>Content Developer</li>
                    <li>Director</li>
                  </ul>
                </div>
              )}
            />
          )}
          required
        />
        <Field
          name="position.organization_override"
          component={ReduxFormCreatableSelect}
          label={<FieldLabel text="Organization" />}
          currentValue={organizationName}
          options={
            allUniqueOrganizations.map(org => ({
              label: org,
              value: org,
            }))
          }
          isCreatable
          formatCreateLabel={(label) => label}
          validate={basicValidate}
          placeholder="Organization..."
        />
        <Field
          name="bio"
          component={RichEditor}
          label={<FieldLabel text="Biography" />}
          maxChars={1000}
          validate={basicValidate}
          id="bio"
        />
        <Field
          name="major_works"
          component={RichEditor}
          label={<FieldLabel text="Major works" optional />}
          maxChars={250}
          id="works"
        />
        <FieldLabel text="Social links" className="mb-2" optional />
        <FieldArray
          name="urls_detailed"
          component={SocialLinks}
        />
        <FieldLabel text="Areas of expertise" className="mb-2" optional />
        <FieldArray
          name="areas_of_expertise"
          component={AreasOfExpertise}
        />
        <ButtonToolbar>
          <Link
            className="btn btn-outline-primary form-cancel-btn"
            to={referrer || '/'}
            disabled={formControlDisabled}
            onClick={cancelStafferInfo}
          >
            Cancel
          </Link>
          <ActionButton
            disabled={formControlDisabled}
            labels={isCreateForm ? {
              default: 'Create',
              pending: 'Creating',
            } : {
              default: 'Save',
              pending: 'Saving',
            }}
            state={isSaving ? 'pending' : 'default'}
          />
        </ButtonToolbar>
      </form>
    </div>
  );
};

BaseStafferForm.propTypes = {
  cancelStafferInfo: PropTypes.func,
  handleSubmit: PropTypes.func.isRequired,
  isSaving: PropTypes.bool,
  isCreateForm: PropTypes.bool,
  organizationName: PropTypes.string,
  pristine: PropTypes.bool.isRequired,
  sourceInfo: PropTypes.shape({
    referrer: PropTypes.string,
  }).isRequired,
  publisherOrganizations: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
  })),
};

BaseStafferForm.defaultProps = {
  cancelStafferInfo: () => {},
  isSaving: false,
  isCreateForm: false,
  organizationName: '',
  publisherOrganizations: [],
};

export default reduxForm({
  form: 'staffer-form',
  onSubmitFail: handleStafferOrCreateFormFail,
})(BaseStafferForm);
export { basicValidate, BaseStafferForm };
