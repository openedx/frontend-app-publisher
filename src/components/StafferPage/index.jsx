import React from 'react';
import PropTypes from 'prop-types';
import { Helmet } from 'react-helmet';

import StafferForm from './StafferForm';
import StatusAlert from '../StatusAlert';
import LoadingSpinner from '../LoadingSpinner';
import PageContainer from '../PageContainer';

class StafferPage extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      startedFetching: false,
    };
    this.handleStafferCreate = this.handleStafferCreate.bind(this);
    this.handleStafferEdit = this.handleStafferEdit.bind(this);
    this.setStartedFetching = this.setStartedFetching.bind(this);
    this.prepareStafferData = this.prepareStafferData.bind(this);
  }

  componentDidMount() {
    this.props.fetchStafferInfo();
    this.setStartedFetching();
  }

  setStartedFetching() {
    this.setState({ startedFetching: true });
  }

  prepareSocialLinks(links) {
    if (!links) {
      return [];
    }

    return links.map(link => (
      ({
        ...link,
        id: link.id || null,
        title: link.title || '',
      })
    ));
  }

  prepareAreasOfExpertise(areasOfExpertise) {
    if (!areasOfExpertise) {
      return [];
    }

    return areasOfExpertise.map(area => ({
      ...area,
      id: area.id || null,
    }));
  }

  preparePosition(position) {
    // If they add an override, we will null out any potential foreign key they used to have
    // so we avoid the situation of a mismatch between having a foreign key to edX, but an
    // override to "New OrgX" and both are present for the person.
    const orgId = position.organization_override ? null : position.organization_id;
    return {
      title: position.title,
      organization: orgId,
      organization_override: position.organization_override,
    };
  }

  prepareStafferData(fieldValues) {
    return {
      ...fieldValues,
      profile_image: fieldValues.profile_image.medium.url,
      areas_of_expertise: this.prepareAreasOfExpertise(fieldValues.areas_of_expertise),
      urls_detailed: this.prepareSocialLinks(fieldValues.urls_detailed),
      position: this.preparePosition(fieldValues.position),
    };
  }

  handleStafferCreate(fieldValues) {
    const {
      createStaffer,
      sourceInfo: { referrer },
    } = this.props;

    const stafferData = this.prepareStafferData(fieldValues);
    createStaffer(stafferData, referrer);
  }

  handleStafferEdit(fieldValues) {
    const {
      editStaffer,
      sourceInfo: { referrer },
    } = this.props;

    const stafferData = this.prepareStafferData(fieldValues);
    if (!stafferData.profile_image.startsWith('data:')) {
      // Only send profile_image if a new one is being uploaded
      delete stafferData.profile_image;
    }
    editStaffer(stafferData, referrer);
  }

  render() {
    const {
      stafferInfo,
      sourceInfo,
    } = this.props;

    if (!stafferInfo) {
      return (
        <StatusAlert
          id="error"
          alertType="danger"
          title="Could not load page"
          message="Could not get instructor information"
        />
      );
    }

    const { startedFetching } = this.state;
    const showForm = (startedFetching && !stafferInfo.isFetching);
    const showSpinner = !showForm;
    const isCreateForm = !this.props.editStaffer;

    const titleText = isCreateForm ? 'Create New Instructor' : 'Edit Instructor';
    const handleSubmit = (isCreateForm
      ? this.handleStafferCreate
      : this.handleStafferEdit);

    const { data, isSaving } = stafferInfo;

    const errorArray = [];

    if (stafferInfo.error) {
      stafferInfo.error.forEach((error, index) => {
        errorArray.push(error);
        if (index < stafferInfo.error.length) {
          errorArray.push(<br />);
        }
      });
    }
    const { referrer } = sourceInfo;
    const organizationName = (data && data.position && data.position.organization_name) || '';

    return (
      <>
        <Helmet>
          <title>{titleText}</title>
        </Helmet>
        <PageContainer>
          { showSpinner && <LoadingSpinner /> }
          { referrer
            && (
            <StatusAlert
              id="sent-from-edit-course-info"
              alertType="info"
              message="The data you entered on the course edit screen is saved. You will return to that page when you have finished updating instructor information."
              dismissible
            />
            )}
          { showForm && (
            <div>
              <h2>{titleText}</h2>
              <hr />
              <StafferForm
                id="create-staffer-form"
                onSubmit={handleSubmit}
                isSaving={isSaving}
                isCreateForm={isCreateForm}
                initialValues={data}
                organizationName={organizationName}
                {...this.props}
              />
              { errorArray.length > 1 && (
                <StatusAlert
                  id="create-staffer-error"
                  alertType="danger"
                  message={errorArray}
                />
              )}
            </div>
          )}
        </PageContainer>
      </>
    );
  }
}

StafferPage.defaultProps = {
  createStaffer: () => {},
  editStaffer: null,
  fetchStafferInfo: () => null,
  stafferInfo: null,
  sourceInfo: {},
};

StafferPage.propTypes = {
  createStaffer: PropTypes.func,
  editStaffer: PropTypes.func,
  fetchStafferInfo: PropTypes.func,
  stafferInfo: PropTypes.shape({
    isFetching: PropTypes.bool,
    isSaving: PropTypes.bool,
    data: PropTypes.shape(),
    error: PropTypes.arrayOf(PropTypes.string),
  }),
  sourceInfo: PropTypes.shape({
    referrer: PropTypes.string,
  }),
};

export default StafferPage;
