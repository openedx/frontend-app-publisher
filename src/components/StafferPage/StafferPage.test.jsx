import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import StafferPage from './index';

describe('StafferPage', () => {
  it('renders html correctly', () => {
    const component = shallow(<StafferPage stafferInfo={{}} />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('renders html correctly when given a referrer', () => {
    const component = shallow(<StafferPage
      stafferInfo={{}}
      sourceInfo={{
        referrer: '/course/11111111-1111-1111-111111111111',
      }}
    />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('renders page correctly with staffer info error', () => {
    const component = shallow(<StafferPage
      createStaffer={() => null}
      stafferInfo={{
        error: ['Fail'],
        isSaving: false,
      }}
    />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('renders page correctly with no stafferInfo', () => {
    const component = shallow(<StafferPage
      createStaffer={() => null}
    />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('renders page correctly while fetching', () => {
    const component = shallow(<StafferPage
      createStaffer={() => null}
      stafferInfo={{
        error: null,
        isSaving: false,
        isFetching: true,
      }}
    />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('renders page correctly while creating', () => {
    const component = shallow(<StafferPage
      createStaffer={() => null}
      stafferInfo={{
        error: null,
        isSaving: true,
      }}
    />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });
});
