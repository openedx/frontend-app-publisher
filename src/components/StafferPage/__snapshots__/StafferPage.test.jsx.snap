// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StafferPage renders html correctly 1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Create New Instructor
    </title>
  </HelmetWrapper>
  <PageContainer
    className=""
    wide={false}
  >
    <div>
      <h2>
        Create New Instructor
      </h2>
      <hr />
      <ReduxForm
        createStaffer={[Function]}
        destroyOnUnmount={true}
        editStaffer={null}
        enableReinitialize={false}
        fetchStafferInfo={[Function]}
        forceUnregisterOnUnmount={false}
        form="staffer-form"
        getFormState={[Function]}
        id="create-staffer-form"
        isCreateForm={true}
        keepDirtyOnReinitialize={false}
        onSubmit={[Function]}
        organizationName=""
        persistentSubmitErrors={false}
        pure={true}
        shouldAsyncValidate={[Function]}
        shouldError={[Function]}
        shouldValidate={[Function]}
        shouldWarn={[Function]}
        sourceInfo={Object {}}
        stafferInfo={Object {}}
        submitAsSideEffect={false}
        touchOnBlur={true}
        touchOnChange={false}
        updateUnregisteredFields={false}
      />
    </div>
  </PageContainer>
</Fragment>
`;

exports[`StafferPage renders html correctly when given a referrer 1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Create New Instructor
    </title>
  </HelmetWrapper>
  <PageContainer
    className=""
    wide={false}
  >
    <StatusAlert
      alertType="info"
      className=""
      dismissible={true}
      iconClassNames={Array []}
      id="sent-from-edit-course-info"
      message="The data you entered on the course edit screen is saved. You will return to that page when you have finished updating instructor information."
      onClose={[Function]}
      title={null}
    />
    <div>
      <h2>
        Create New Instructor
      </h2>
      <hr />
      <ReduxForm
        createStaffer={[Function]}
        destroyOnUnmount={true}
        editStaffer={null}
        enableReinitialize={false}
        fetchStafferInfo={[Function]}
        forceUnregisterOnUnmount={false}
        form="staffer-form"
        getFormState={[Function]}
        id="create-staffer-form"
        isCreateForm={true}
        keepDirtyOnReinitialize={false}
        onSubmit={[Function]}
        organizationName=""
        persistentSubmitErrors={false}
        pure={true}
        shouldAsyncValidate={[Function]}
        shouldError={[Function]}
        shouldValidate={[Function]}
        shouldWarn={[Function]}
        sourceInfo={
          Object {
            "referrer": "/course/11111111-1111-1111-111111111111",
          }
        }
        stafferInfo={Object {}}
        submitAsSideEffect={false}
        touchOnBlur={true}
        touchOnChange={false}
        updateUnregisteredFields={false}
      />
    </div>
  </PageContainer>
</Fragment>
`;

exports[`StafferPage renders page correctly while creating 1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Create New Instructor
    </title>
  </HelmetWrapper>
  <PageContainer
    className=""
    wide={false}
  >
    <div>
      <h2>
        Create New Instructor
      </h2>
      <hr />
      <ReduxForm
        createStaffer={[Function]}
        destroyOnUnmount={true}
        editStaffer={null}
        enableReinitialize={false}
        fetchStafferInfo={[Function]}
        forceUnregisterOnUnmount={false}
        form="staffer-form"
        getFormState={[Function]}
        id="create-staffer-form"
        isCreateForm={true}
        isSaving={true}
        keepDirtyOnReinitialize={false}
        onSubmit={[Function]}
        organizationName=""
        persistentSubmitErrors={false}
        pure={true}
        shouldAsyncValidate={[Function]}
        shouldError={[Function]}
        shouldValidate={[Function]}
        shouldWarn={[Function]}
        sourceInfo={Object {}}
        stafferInfo={
          Object {
            "error": null,
            "isSaving": true,
          }
        }
        submitAsSideEffect={false}
        touchOnBlur={true}
        touchOnChange={false}
        updateUnregisteredFields={false}
      />
    </div>
  </PageContainer>
</Fragment>
`;

exports[`StafferPage renders page correctly while fetching 1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Create New Instructor
    </title>
  </HelmetWrapper>
  <PageContainer
    className=""
    wide={false}
  >
    <LoadingSpinner
      message="Loadingâ€¦"
    />
  </PageContainer>
</Fragment>
`;

exports[`StafferPage renders page correctly with no stafferInfo 1`] = `
<StatusAlert
  alertType="danger"
  className=""
  dismissible={false}
  iconClassNames={Array []}
  id="error"
  message="Could not get instructor information"
  onClose={[Function]}
  title="Could not load page"
/>
`;

exports[`StafferPage renders page correctly with staffer info error 1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Create New Instructor
    </title>
  </HelmetWrapper>
  <PageContainer
    className=""
    wide={false}
  >
    <div>
      <h2>
        Create New Instructor
      </h2>
      <hr />
      <ReduxForm
        createStaffer={[Function]}
        destroyOnUnmount={true}
        editStaffer={null}
        enableReinitialize={false}
        fetchStafferInfo={[Function]}
        forceUnregisterOnUnmount={false}
        form="staffer-form"
        getFormState={[Function]}
        id="create-staffer-form"
        isCreateForm={true}
        isSaving={false}
        keepDirtyOnReinitialize={false}
        onSubmit={[Function]}
        organizationName=""
        persistentSubmitErrors={false}
        pure={true}
        shouldAsyncValidate={[Function]}
        shouldError={[Function]}
        shouldValidate={[Function]}
        shouldWarn={[Function]}
        sourceInfo={Object {}}
        stafferInfo={
          Object {
            "error": Array [
              "Fail",
            ],
            "isSaving": false,
          }
        }
        submitAsSideEffect={false}
        touchOnBlur={true}
        touchOnChange={false}
        updateUnregisteredFields={false}
      />
      <StatusAlert
        alertType="danger"
        className=""
        dismissible={false}
        iconClassNames={Array []}
        id="create-staffer-error"
        message={
          Array [
            "Fail",
            <br />,
          ]
        }
        onClose={[Function]}
        title={null}
      />
    </div>
  </PageContainer>
</Fragment>
`;
